common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/CustomerInfoResponseHandler.kt

```diff
@@ -0,0 +1,104 @@
+package com.revenuecat.purchases.common.offlineentitlements
+
+import com.google.gson.internal.bind.util.ISO8601Utils
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.VerificationResult
+import com.revenuecat.purchases.common.AppConfig
+import com.revenuecat.purchases.common.CustomerInfoFactory
+import com.revenuecat.purchases.common.DateProvider
+import com.revenuecat.purchases.common.DefaultDateProvider
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.strings.CustomerInfoStrings.COMPUTING_OFFLINE_CUSTOMER_INFO_FAILED
+import com.revenuecat.purchases.utils.Iso8601Utils
+import org.json.JSONObject
+import java.util.Date
+
+class CustomerInfoResponseHandler(
+    private val purchasedProductsFetcher: PurchasedProductsFetcher,
+    private val dateProvider: DateProvider = DefaultDateProvider(),
+    private val appConfig: AppConfig
+) {
+
+    fun computeOfflineCustomerInfo(
+        appUserID: String,
+        onSuccess: (CustomerInfo) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        purchasedProductsFetcher.queryPurchasedProducts(
+            appUserID,
+            onSuccess = { purchasedProducts ->
+                val jsonObject = JSONObject()
+                val requestDate = dateProvider.now
+                jsonObject.apply {
+                    val formattedDate = Iso8601Utils.format(requestDate)
+                    put("request_date", formattedDate)
+                    put("request_date_ms", requestDate.time)
+                    put("subscriber", JSONObject().apply {
+                        put("original_app_user_id", appUserID)
+                        put("original_application_version", "1.0")
+                        put("entitlements", generateEntitlementsResponse(purchasedProducts))
+                        put("first_seen", formattedDate)
+                        val originalPurchaseDate = calculateOriginalPurchaseDate(purchasedProducts)
+                        put("original_purchase_date", originalPurchaseDate)
+                        put("non_subscriptions", JSONObject()) // TODO in another PR
+                        put("subscriptions", generateSubscriptions(purchasedProducts))
+                        put("management_url", "https://play.google.com/store/account/subscriptions")
```

Comment:
It's okay if it has to be in a different PR, but it'd be great to share the strings from here [and here](https://github.com/RevenueCat/purchases-android/blob/703750ecc570eb633f6874a1552e225eaf13adb5/common/src/main/java/com/revenuecat/purchases/common/EntitlementInfoFactories.kt#L31), so that if we update one in the future we don't break this and never find out